src/test/java/test/factory/github2428/Reporter.java

```diff
@@ -0,0 +1,19 @@
+package test.factory.github2428;
+
+import org.testng.IReporter;
+import org.testng.ISuite;
+import org.testng.xml.XmlSuite;
+
+import java.util.List;
+
+public class Reporter implements IReporter {
```

Comment:
This test is actually not validating anything apart from it just doing print. Can we please enhance this to be a proper test that relies on assertions to really check if the configuration method have real different objects ?